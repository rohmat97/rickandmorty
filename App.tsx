/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 *
 * @format
 */

import React from "react";
import { StatusBar } from "react-native/types";
import RootNavigation from "./src/navigation";

function App() {
  return <RootNavigation />;
}

export default App;
