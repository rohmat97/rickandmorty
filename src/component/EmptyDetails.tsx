import { Image, StyleSheet, Text, View } from "react-native";
import React from "react";

const EmptyDetails = () => {
  return (
    <View style={{ flex: 1 }}>
      <View
        style={{
          width: "100%",
          height: 375,
          backgroundColor: "#C4C4C4",
        }}
      />
      <View style={{ margin: 16 }}>
        <View style={styles.headerTextBlur} />
        <View style={styles.content} />
        <View style={styles.line} />
      </View>
    </View>
  );
};

export default EmptyDetails;

const styles = StyleSheet.create({
  headerTextBlur: {
    marginTop: 10,
    backgroundColor: "#C4C4C4",
    height: 16,
    width: 132,
  },
  content: {
    marginTop: 18,
    backgroundColor: "#C4C4C4",
    height: 74,
    width: "100%",
    borderRadius: 4,
  },
  line: {
    marginTop: 36,
    alignSelf: "center",
    backgroundColor: "#C4C4C4",
    height: 1,
    width: 200,
  },
});
