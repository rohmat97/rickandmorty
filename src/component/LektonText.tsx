import { Text, TextProps } from "react-native";
import React, { FunctionComponent } from "react";

const LektonText = (
  props: JSX.IntrinsicAttributes &
    JSX.IntrinsicClassAttributes<Text> &
    Readonly<TextProps>
) => (
  <Text {...props} style={[{ fontFamily: "Lekton-Bold" }, props.style]}>
    {props.children}
  </Text>
);

export default LektonText;
