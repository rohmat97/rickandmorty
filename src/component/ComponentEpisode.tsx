import { StyleSheet, Text, View } from "react-native";
import React, { FunctionComponent, useState } from "react";
import { LektonBold, LektonNormal } from "./LektonText";
import axios from "axios";

const ComponentEpisode: FunctionComponent<any> = ({ item }) => {
  const [episodes, setepisode] = useState();
  axios.get(item).then((succ: any) =>
    setTimeout(() => {
      setepisode(succ.data);
    }, 1000)
  );
  //   console.log("episode", episode);
  if (episodes) {
    const { name, episode, air_date }: any = episodes;
    if (name && episode && air_date) {
      return (
        <View style={{ height: 68, paddingTop: 16 }}>
          <LektonBold style={{ color: "black", fontSize: 16 }}>
            {name} - {episode}
          </LektonBold>
          <LektonNormal style={{ fontSize: 12 }}>{air_date}</LektonNormal>
        </View>
      );
    }
    return <></>;
  }
  return (
    <View style={styles.container}>
      <View style={styles.menuContent}>
        <View style={styles.headerTextBlur} />
        <View style={styles.descTextBlur} />
      </View>
    </View>
  );
};

export default ComponentEpisode;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: "row",
    padding: 16,
    // borderWidth: 1,
  },
  menuContent: {
    flex: 1,
    marginLeft: 16,
    height: 40,
    justifyContent: "space-between",
  },
  imageBlur: {
    backgroundColor: "#C4C4C4",
    height: 40,
    width: 40,
  },
  headerTextBlur: {
    backgroundColor: "#C4C4C4",
    height: 16,
    width: 132,
  },
  descTextBlur: {
    backgroundColor: "#C4C4C4",
    height: 16,
    width: 85,
  },
});
