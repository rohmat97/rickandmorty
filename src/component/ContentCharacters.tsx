import { Image, StyleSheet, TouchableOpacity, View } from "react-native";
import React, { FunctionComponent } from "react";
import { LektonBold, LektonNormal } from "./LektonText";

const ContentCharacters: FunctionComponent<any> = ({ item, navigation }) => {
  const { name, image, species } = item;

  const moveToDetails = () => {
    navigation.navigate("Details", {
      data: item,
    });
  };

  return (
    <TouchableOpacity
      testID="testNavigateDetail"
      onPress={() => moveToDetails()}
    >
      <View style={styles.container}>
        <Image source={{ uri: image }} style={styles.imageBlur} />
        <View style={styles.menuContent}>
          <LektonBold style={{ color: "black" }}>{name}</LektonBold>
          <LektonNormal>{species}</LektonNormal>
        </View>
      </View>
    </TouchableOpacity>
  );
};

export default ContentCharacters;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: "row",
    padding: 16,
    // borderWidth: 1,
  },
  menuContent: {
    flex: 1,
    marginLeft: 16,
    height: 40,
    justifyContent: "space-between",
  },
  imageBlur: {
    backgroundColor: "#C4C4C4",
    height: 40,
    width: 40,
  },
  headerTextBlur: {
    backgroundColor: "#C4C4C4",
    height: 16,
    width: 132,
  },
  descTextBlur: {
    backgroundColor: "#C4C4C4",
    height: 16,
    width: 85,
  },
});
