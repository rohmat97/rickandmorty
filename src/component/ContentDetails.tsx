import { FlatList, Image, ScrollView, StyleSheet, View } from "react-native";
import React, { FunctionComponent } from "react";
import { LektonBold, LektonNormal } from "./LektonText";
import ComponentEpisode from "./ComponentEpisode";

const ContentDetails: FunctionComponent<any> = ({ detailData }) => {
  const { name, image, species, gender, episode, status } = detailData;
  return (
    <ScrollView>
      <View style={{ flex: 1 }}>
        <Image
          source={{ uri: image }}
          style={styles.image}
          resizeMode="cover"
        />
        <View style={{ margin: 16 }}>
          <LektonBold style={styles.header}>{name}</LektonBold>
          <View style={styles.content}>
            <LektonNormal style={{ color: "white" }}>
              spesies: {species}
            </LektonNormal>
            <LektonNormal style={{ color: "white" }}>
              gender: {gender}
            </LektonNormal>
            <LektonNormal style={{ color: "white" }}>
              status: {status}
            </LektonNormal>
          </View>
          <View style={styles.line} />
          <LektonBold style={[styles.header, { marginTop: 36, fontSize: 20 }]}>
            Episodes
          </LektonBold>
          <FlatList
            data={episode || []}
            renderItem={({ item }) => <ComponentEpisode item={item} />}
            scrollEnabled={false}
          />
        </View>
      </View>
    </ScrollView>
  );
};

export default ContentDetails;

const styles = StyleSheet.create({
  headerTextBlur: {
    marginTop: 10,
    backgroundColor: "#C4C4C4",
    height: 16,
    width: 132,
  },
  content: {
    marginTop: 18,
    backgroundColor: "#EF5DA8",
    height: 74,
    width: "100%",
    borderRadius: 4,
    paddingLeft: 8,
  },
  line: {
    marginTop: 36,
    alignSelf: "center",
    backgroundColor: "#C4C4C4",
    height: 1,
    width: 200,
  },
  header: {
    color: "black",
    fontSize: 20,
    // fontWeight: "700",
  },
  image: {
    width: "100%",
    height: 375,
    backgroundColor: "#C4C4C4",
  },
});
