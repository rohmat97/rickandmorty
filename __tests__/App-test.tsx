/**
 * @format
 */

import "react-native";
import React from "react";

// Note: test renderer must be required after react-native.
import renderer, { act, create } from "react-test-renderer";
import { LektonBold, LektonNormal } from "../src/component/LektonText";
import EmptyContent from "../src/component/EmptyContent";
import EmptyDetails from "../src/component/EmptyDetails";
import ContentDetails from "../src/component/ContentDetails";
import ContentCharacters from "../src/component/ContentCharacters";
import ComponentEpisode from "../src/component/ComponentEpisode";
import App from "../App";
import { Image, View } from "react-native";

const sampleResponse = {
  info: {
    count: 826,
    pages: 42,
    next: "https://rickandmortyapi.com/api/character/?page=20",
    prev: "https://rickandmortyapi.com/api/character/?page=18",
  },
  results: [
    {
      id: 361,
      name: "Toxic Rick",
      status: "Dead",
      species: "Humanoid",
      type: "Rick's Toxic Side",
      gender: "Male",
      origin: {
        name: "Alien Spa",
        url: "https://rickandmortyapi.com/api/location/64",
      },
      location: {
        name: "Earth",
        url: "https://rickandmortyapi.com/api/location/20",
      },
      image: "https://rickandmortyapi.com/api/character/avatar/361.jpeg",
      episode: ["https://rickandmortyapi.com/api/episode/27"],
      url: "https://rickandmortyapi.com/api/character/361",
      created: "2018-01-10T18:20:41.703Z",
    },
    // ...
  ],
};

it("test render component text Lekton Bold", () => {
  const tree = renderer.create(<LektonBold>LektonBold</LektonBold>).toJSON();
  expect(tree).toMatchSnapshot();
});
it("test render component text Lekton Regular", () => {
  const tree = renderer
    .create(<LektonNormal>LektonNormal</LektonNormal>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
it("test render component empty content", () => {
  const tree = renderer.create(<EmptyContent />).toJSON();
  expect(tree).toMatchSnapshot();
});
it("test render component empty content detail", () => {
  const tree = renderer.create(<EmptyDetails />).toJSON();
  expect(tree).toMatchSnapshot();
});
it("test render component ContentCharacters", () => {
  const tree = renderer
    .create(<ContentCharacters item={sampleResponse.results} navigation={{}} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
it("test render component ContentDetails", () => {
  const tree = renderer
    .create(<ContentDetails detailData={sampleResponse.results} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("test render component ComponentEpisode", () => {
  const tree = renderer
    .create(<ComponentEpisode item={sampleResponse.results.episode} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
