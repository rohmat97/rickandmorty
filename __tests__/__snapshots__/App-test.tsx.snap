// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test render component ComponentEpisode 1`] = `
<View
  style={
    {
      "flex": 1,
      "flexDirection": "row",
      "padding": 16,
    }
  }
>
  <View
    style={
      {
        "flex": 1,
        "height": 40,
        "justifyContent": "space-between",
        "marginLeft": 16,
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "#C4C4C4",
          "height": 16,
          "width": 132,
        }
      }
    />
    <View
      style={
        {
          "backgroundColor": "#C4C4C4",
          "height": 16,
          "width": 85,
        }
      }
    />
  </View>
</View>
`;

exports[`test render component ContentCharacters 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
  testID="testNavigateDetail"
>
  <View
    style={
      {
        "flex": 1,
        "flexDirection": "row",
        "padding": 16,
      }
    }
  >
    <Image
      source={
        {
          "uri": undefined,
        }
      }
      style={
        {
          "backgroundColor": "#C4C4C4",
          "height": 40,
          "width": 40,
        }
      }
    />
    <View
      style={
        {
          "flex": 1,
          "height": 40,
          "justifyContent": "space-between",
          "marginLeft": 16,
        }
      }
    >
      <Text
        style={
          [
            {
              "fontFamily": "Lekton-Bold",
            },
            {
              "color": "black",
            },
          ]
        }
      />
      <Text
        style={
          [
            {
              "fontFamily": "Lekton-Regular",
            },
            undefined,
          ]
        }
      />
    </View>
  </View>
</View>
`;

exports[`test render component ContentDetails 1`] = `
<RCTScrollView>
  <View>
    <View
      style={
        {
          "flex": 1,
        }
      }
    >
      <Image
        resizeMode="cover"
        source={
          {
            "uri": undefined,
          }
        }
        style={
          {
            "backgroundColor": "#C4C4C4",
            "height": 375,
            "width": "100%",
          }
        }
      />
      <View
        style={
          {
            "margin": 16,
          }
        }
      >
        <Text
          style={
            [
              {
                "fontFamily": "Lekton-Bold",
              },
              {
                "color": "black",
                "fontSize": 20,
              },
            ]
          }
        />
        <View
          style={
            {
              "backgroundColor": "#EF5DA8",
              "borderRadius": 4,
              "height": 74,
              "marginTop": 18,
              "paddingLeft": 8,
              "width": "100%",
            }
          }
        >
          <Text
            style={
              [
                {
                  "fontFamily": "Lekton-Regular",
                },
                {
                  "color": "white",
                },
              ]
            }
          >
            spesies: 
          </Text>
          <Text
            style={
              [
                {
                  "fontFamily": "Lekton-Regular",
                },
                {
                  "color": "white",
                },
              ]
            }
          >
            gender: 
          </Text>
          <Text
            style={
              [
                {
                  "fontFamily": "Lekton-Regular",
                },
                {
                  "color": "white",
                },
              ]
            }
          >
            status: 
          </Text>
        </View>
        <View
          style={
            {
              "alignSelf": "center",
              "backgroundColor": "#C4C4C4",
              "height": 1,
              "marginTop": 36,
              "width": 200,
            }
          }
        />
        <Text
          style={
            [
              {
                "fontFamily": "Lekton-Bold",
              },
              [
                {
                  "color": "black",
                  "fontSize": 20,
                },
                {
                  "fontSize": 20,
                  "marginTop": 36,
                },
              ],
            ]
          }
        >
          Episodes
        </Text>
        <RCTScrollView
          data={[]}
          getItem={[Function]}
          getItemCount={[Function]}
          keyExtractor={[Function]}
          onContentSizeChange={[Function]}
          onLayout={[Function]}
          onMomentumScrollBegin={[Function]}
          onMomentumScrollEnd={[Function]}
          onScroll={[Function]}
          onScrollBeginDrag={[Function]}
          onScrollEndDrag={[Function]}
          removeClippedSubviews={false}
          renderItem={[Function]}
          scrollEnabled={false}
          scrollEventThrottle={50}
          stickyHeaderIndices={[]}
          viewabilityConfigCallbackPairs={[]}
        >
          <View />
        </RCTScrollView>
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`test render component empty content 1`] = `
<View
  style={
    {
      "flex": 1,
      "flexDirection": "row",
      "padding": 16,
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "#C4C4C4",
        "height": 40,
        "width": 40,
      }
    }
  />
  <View
    style={
      {
        "flex": 1,
        "height": 40,
        "justifyContent": "space-between",
        "marginLeft": 16,
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "#C4C4C4",
          "height": 16,
          "width": 132,
        }
      }
    />
    <View
      style={
        {
          "backgroundColor": "#C4C4C4",
          "height": 16,
          "width": 85,
        }
      }
    />
  </View>
</View>
`;

exports[`test render component empty content detail 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "#C4C4C4",
        "height": 375,
        "width": "100%",
      }
    }
  />
  <View
    style={
      {
        "margin": 16,
      }
    }
  >
    <View
      style={
        {
          "backgroundColor": "#C4C4C4",
          "height": 16,
          "marginTop": 10,
          "width": 132,
        }
      }
    />
    <View
      style={
        {
          "backgroundColor": "#C4C4C4",
          "borderRadius": 4,
          "height": 74,
          "marginTop": 18,
          "width": "100%",
        }
      }
    />
    <View
      style={
        {
          "alignSelf": "center",
          "backgroundColor": "#C4C4C4",
          "height": 1,
          "marginTop": 36,
          "width": 200,
        }
      }
    />
  </View>
</View>
`;

exports[`test render component text Lekton Bold 1`] = `
<Text
  style={
    [
      {
        "fontFamily": "Lekton-Bold",
      },
      undefined,
    ]
  }
>
  LektonBold
</Text>
`;

exports[`test render component text Lekton Regular 1`] = `
<Text
  style={
    [
      {
        "fontFamily": "Lekton-Regular",
      },
      undefined,
    ]
  }
>
  LektonNormal
</Text>
`;
